src/server/game/Entities/Player/Player.cpp

```diff
@@ -10990,47 +10992,78 @@ InventoryResult Player::CanStoreItems(Item** items, int count, uint32* itemLimit
     memset(keyringCounts, 0, sizeof(uint32) * (KEYRING_SLOT_END - KEYRING_SLOT_START));
     memset(currencyCounts, 0, sizeof(uint32) * (CURRENCYTOKEN_SLOT_END - CURRENCYTOKEN_SLOT_START));
 
+    // Item pointers
+    Item* inventoryPointers[INVENTORY_SLOT_ITEM_END - INVENTORY_SLOT_ITEM_START];
+    Item* bagPointers[INVENTORY_SLOT_BAG_END - INVENTORY_SLOT_BAG_START][MAX_BAG_SIZE];
+    Item* keyringPointers[KEYRING_SLOT_END - KEYRING_SLOT_START];
+    Item* currencyPointers[CURRENCYTOKEN_SLOT_END - CURRENCYTOKEN_SLOT_START];
+
+    memset(inventoryPointers, 0, sizeof(Item*) * (INVENTORY_SLOT_ITEM_END - INVENTORY_SLOT_ITEM_START));
```

Comment:
could there be a more modern C++ way to default initialize all those local arrays to nullptr ?

Reply:
I could change it to be along the lines of `Item* keyringPointers[KEYRING_SLOT_END - KEYRING_SLOT_START] = {}`. This should remove the need for memset.